apply/apply.go

```diff
@@ -204,7 +239,19 @@ func applyEnvs(
 			}
 
 			if componentPlan.ModuleSource != nil {
-				e := applyModuleInvocation(fs, path, *componentPlan.ModuleSource, componentPlan.ModuleName, templates.Templates.ModuleInvocation, commonBox)
+				downloader := util.MakeDownloader(*componentPlan.ModuleSource)
+				if oauth {
+					type HTTPAuth struct {
+						GithubToken string `required:"true"`
+					}
+					var httpAuth HTTPAuth
+					err := envconfig.Process("fogg", &httpAuth)
+					if err != nil {
+						return errs.WrapUser(err, "unable to get env FOGG_GITHUBTOKEN with the --oauth flag")
+					}
```

Comment:
should you check this much earlier (like at the cobra validation stage) so that you fail early if the env is not set when the flag is?